clients/src/main/java/org/apache/kafka/common/protocol/ProtoUtils.java

```diff
@@ -27,7 +27,7 @@ private static Schema schemaFor(Schema[][] schemas, int apiKey, int version) {
         if (apiKey < 0 || apiKey > schemas.length)
             throw new IllegalArgumentException("Invalid api key: " + apiKey);
         Schema[] versions = schemas[apiKey];
-        if (version < 0 || version > versions.length)
+        if (version < 0 || version >= versions.length)
```

Comment:
I don't have the full source open in front of me, is there a convenient getter for `latest version` available already?  As far as performance, I believe an array length check is about as fast as it gets short of a constant value.  For readability though I think `version > versions.latest` would be more clear if that is an available thing.